Before: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
After: 2f51a09e8ec364cec8a95acb206c9315cefb0830081c7eb6eb073472e379dc67
Patch: 9844fb8ddaef5d1b90f0f298eaa3452381b6c548f2df8a748fd96f1823ae176d
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ .gitignore	2012-07-22 20:44:49.248917603 +1000
@@ -0,0 +1,17 @@
+~*
+.~*
+*.bak
+*.log
+*.tmp
+*.out
+*.err
+*.pid
+.*.swp
+.*.swo
+.*.swn
+
+.settings
+.metadata
+.idea
+target
+sloccount.sc
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ .rspec	2012-07-22 20:44:49.248917603 +1000
@@ -0,0 +1,8 @@
+--color
+
+--order random
+#--order random:12321 # Use a specific seed value
+
+#--format progress
+--format documentation
+
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ .simplecov	2012-07-22 20:44:49.252250904 +1000
@@ -0,0 +1,14 @@
+SimpleCov.start do
+  # project_name 'My Awesome Project'
+
+  # add_group 'Models', 'app/model'
+  # add_group 'Plugins', '(app|lib)/plugins'
+
+  # Remove test code from coverage
+  add_filter 'test'
+
+  # Add files that don't get required to coverage too
+  add_files_to_coverage_at_exit '{app,lib}/**/*.rb'
+end
+
+# vim:ft=ruby et ts=2 sw=2:
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ .yardopts	2012-07-22 20:44:49.248917603 +1000
@@ -0,0 +1,6 @@
+--db         target/yardoc.db
+--output-dir target/doc
+--markup     markdown
+--readme     README.md
+--files      CHANGELOG.md
+{app,lib}/**/*.rb
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ CHANGELOG.md	2012-07-22 20:44:49.252250904 +1000
@@ -0,0 +1,6 @@
+My Project: Changelog
+=====================
+
+## 0.0.1 (WIP)
+
+Initial version.
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ Gemfile	2012-07-22 20:44:49.248917603 +1000
@@ -0,0 +1,11 @@
+source :rubygems
+
+# Load Gemfile.corvid
+eval File.read(__FILE__ + '.corvid')
+
+# Parser for Markdown documentation
+group :doc do
+  gem 'rdiscount', platforms: :mri
+  gem 'kramdown', platforms: :jruby
+end
+
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ Gemfile.corvid	2012-07-22 20:44:49.252250904 +1000
@@ -0,0 +1,34 @@
+# vim:ft=ruby et ts=2 sw=2:
+header "# This file is generated by corvid update:deps"
+
+if options[:use_corvid_gem]
+  gem 'corvid', ">= #{::Corvid::VERSION}"
+else
+  gem 'corvid', path: CORVID_ROOT
+end
+gem 'rake', group: :rake
+
+group :test do
+  gem 'ci_reporter', require: false
+  gem 'simplecov', '>= 0.6.4', require: false
+end if Dir.exists?('test')
+
+group :test_unit do
+  gem 'minitest'
+  gem 'guard-minitest', '>= 0.5.0', require: false
+  gem 'turn', '>= 0.9.4'
+end if Dir.exists?('test/unit')
+
+group :test_spec do
+  gem 'rspec'
+  gem 'guard-rspec', require: false
+end if Dir.exists?('test/spec')
+
+group :doc do
+  gem 'yard'
+end
+
+group :debug do
+  gem 'debugger', platforms: :mri
+end
+
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ Gemfile.raven	2012-07-22 20:44:49.485582560 +1000
@@ -0,0 +1,38 @@
+# vim: ft=ruby et ts=2 sw=2 :
+header "# This file is generated by raven update:deps"
+footer "# v\im: ft=ruby et ts=2 sw=2 :"
+
+if options[:use_raven_gem]
+  gem 'raven', ">= #{::Raven::VERSION}"
+else
+  gem 'raven', path: RAVEN_ROOT
+end
+gem 'rake', group: :rake
+
+group :test do
+  gem 'simplecov'
+end if Dir.exists?('test')
+
+group :test_unit do
+  gem 'minitest'
+  gem 'guard-minitest', '>= 0.5.0'
+end if Dir.exists?('test/unit')
+
+group :test_spec do
+  gem 'rspec'
+  gem 'guard-rspec'
+end if Dir.exists?('test/spec')
+
+group :ci do
+  gem 'ci_reporter', require: false
+end
+
+group :doc do
+  gem 'yard'
+  gem 'rdiscount' # TODO aaaaaaaaaahhhhhhhhhh
+end
+
+group :debug do
+  gem 'debugger', platforms: :mri
+end
+
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ Guardfile	2012-07-22 20:44:49.252250904 +1000
@@ -0,0 +1,44 @@
+require 'corvid/guard'
+
+########################################################################################################################
+# test/unit
+
+group :unit do
+  guard 'minitest', test_folders: 'test/unit', test_file_patterns: '*_test.rb' do
+
+    # Src files
+    watch(%r'^lib/(.+)\.rb$') {|m| "test/unit/#{m[1]}_test.rb"}
+
+    # Each test
+    watch(%r'^test/unit/.+_test\.rb$')
+
+    if bulk?
+      # Test stuff affecting everything
+      watch(%r'^test/bootstrap/(?:all|unit).rb') {"test/unit"}
+      watch(%r'^test/helpers/.+$')               {"test/unit"}
+      watch(%r'^test/factories/.+$')             {"test/unit"}
+    end
+  end
+end if Dir.exists?('test/unit')
+
+########################################################################################################################
+# test/spec
+
+rspec_options= read_rspec_options(File.dirname __FILE__)
+group :spec do
+  guard 'rspec', binstubs: true, spec_paths: ['test/spec'], cli: rspec_options, all_on_start: false, all_after_pass: false do
+
+    # Src files
+    watch(%r'^lib/(.+)\.rb$') {|m| "test/spec/#{m[1]}_spec.rb"}
+
+    # Each spec
+    watch(%r'^test/spec/.+_spec\.rb$')
+
+    if bulk?
+      # Test stuff affecting everything
+      watch(%r'^test/bootstrap/(?:all|spec).rb') {"test/spec"}
+      watch(%r'^test/helpers/.+$')               {"test/spec"}
+      watch(%r'^test/factories/.+$')             {"test/spec"}
+    end
+  end
+end if Dir.exists?('test/spec')
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ README.md	2012-07-22 20:44:49.248917603 +1000
@@ -0,0 +1,6 @@
+# My Project [![Build Status](https://secure.travis-ci.org/myusername/myproject.png?branch=master)](http://travis-ci.org/myusername/myproject) [![Dependency Status](https://gemnasium.com/myusername/myproject.png)](https://gemnasium.com/myusername/myproject)
+
+For the above Travis CI and Gemnasium buttons to work you must replace:
+* `myusername` with your Github username.
+* `myproject` with your project's name on Github.
+
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ Rakefile	2012-07-22 20:44:49.248917603 +1000
@@ -0,0 +1,6 @@
+raise "Rake must not be run directly. Either run via Bundler (bundle exec rake) or via bin/rake." unless defined?(Bundler)
+APP_ROOT ||= File.expand_path(File.dirname(__FILE__))
+require 'corvid/rake/tasks'
+
+# Set default task to test
+task :default => :test
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ bin/guard	2012-07-22 20:44:49.252250904 +1000
@@ -0,0 +1,16 @@
+#!/usr/bin/env ruby
+#
+# This file was generated by Bundler.
+#
+# The application 'guard' is installed as part of a gem, and
+# this file is here to facilitate running it.
+#
+
+require 'pathname'
+ENV['BUNDLE_GEMFILE'] ||= File.expand_path("../../Gemfile",
+  Pathname.new(__FILE__).realpath)
+
+require 'rubygems'
+require 'bundler/setup'
+
+load Gem.bin_path('guard', 'guard')
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ bin/rake	2012-07-22 20:44:49.252250904 +1000
@@ -0,0 +1,16 @@
+#!/usr/bin/env ruby
+#
+# This file was generated by Bundler.
+#
+# The application 'rake' is installed as part of a gem, and
+# this file is here to facilitate running it.
+#
+
+require 'pathname'
+ENV['BUNDLE_GEMFILE'] ||= File.expand_path("../../Gemfile",
+  Pathname.new(__FILE__).realpath)
+
+require 'rubygems'
+require 'bundler/setup'
+
+load Gem.bin_path('rake', 'rake')
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ bin/rspec	2012-07-22 20:44:49.248917603 +1000
@@ -0,0 +1,16 @@
+#!/usr/bin/env ruby
+#
+# This file was generated by Bundler.
+#
+# The application 'rspec' is installed as part of a gem, and
+# this file is here to facilitate running it.
+#
+
+require 'pathname'
+ENV['BUNDLE_GEMFILE'] ||= File.expand_path("../../Gemfile",
+  Pathname.new(__FILE__).realpath)
+
+require 'rubygems'
+require 'bundler/setup'
+
+load Gem.bin_path('rspec-core', 'rspec')
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ bin/yard	2012-07-22 20:44:49.252250904 +1000
@@ -0,0 +1,16 @@
+#!/usr/bin/env ruby
+#
+# This file was generated by Bundler.
+#
+# The application 'yard' is installed as part of a gem, and
+# this file is here to facilitate running it.
+#
+
+require 'pathname'
+ENV['BUNDLE_GEMFILE'] ||= File.expand_path("../../Gemfile",
+  Pathname.new(__FILE__).realpath)
+
+require 'rubygems'
+require 'bundler/setup'
+
+load Gem.bin_path('yard', 'yard')
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ bin/yardoc	2012-07-22 20:44:49.248917603 +1000
@@ -0,0 +1,16 @@
+#!/usr/bin/env ruby
+#
+# This file was generated by Bundler.
+#
+# The application 'yardoc' is installed as part of a gem, and
+# this file is here to facilitate running it.
+#
+
+require 'pathname'
+ENV['BUNDLE_GEMFILE'] ||= File.expand_path("../../Gemfile",
+  Pathname.new(__FILE__).realpath)
+
+require 'rubygems'
+require 'bundler/setup'
+
+load Gem.bin_path('yard', 'yardoc')
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ bin/yri	2012-07-22 20:44:49.252250904 +1000
@@ -0,0 +1,16 @@
+#!/usr/bin/env ruby
+#
+# This file was generated by Bundler.
+#
+# The application 'yri' is installed as part of a gem, and
+# this file is here to facilitate running it.
+#
+
+require 'pathname'
+ENV['BUNDLE_GEMFILE'] ||= File.expand_path("../../Gemfile",
+  Pathname.new(__FILE__).realpath)
+
+require 'rubygems'
+require 'bundler/setup'
+
+load Gem.bin_path('yard', 'yri')
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ test/bootstrap/all.rb	2012-07-22 20:44:49.252250904 +1000
@@ -0,0 +1,11 @@
+# encoding: utf-8
+APP_ROOT = File.expand_path('../../..',__FILE__)
+require 'rubygems'
+require 'bundler/setup'
+require 'corvid/test/bootstrap/all'
+
+Bundler.require :default
+
+# Load test helpers
+Dir.glob("#{APP_ROOT}/test/helpers/**/*.rb") {|f| require f}
+
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ test/bootstrap/all.rb.tt	2012-07-22 20:44:49.532248890 +1000
@@ -0,0 +1,2 @@
+# encoding: utf-8
+APP_DIR = File.expand_path('../../..',__FILE__)
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ test/bootstrap/spec.rb	2012-07-22 20:44:49.252250904 +1000
@@ -0,0 +1,5 @@
+# encoding: utf-8
+$coverage_name = 'specs'
+require_relative 'all'
+require 'corvid/test/bootstrap/spec'
+
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ test/bootstrap/unit.rb	2012-07-22 20:44:49.252250904 +1000
@@ -0,0 +1,4 @@
+# encoding: utf-8
+$coverage_name = 'unit tests'
+require_relative 'all'
+require 'corvid/test/bootstrap/unit'
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ test/bootstrap/unit.rb.tt	2012-07-22 20:44:49.532248890 +1000
@@ -0,0 +1,3 @@
+# encoding: utf-8
+require_relative 'all'
+
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ test/spec/%src%_spec.rb.tt	2012-07-22 20:44:49.252250904 +1000
@@ -0,0 +1,7 @@
+# encoding: utf-8
+require_relative '<%= bootstrap_dir %>/spec'
+require '<%= src %>'
+
+describe <%= subject %> do
+  # TODO
+end
--- /dev/null	2012-07-22 20:29:06.375709578 +1000
+++ test/unit/%src%_test.rb.tt	2012-07-22 20:44:49.252250904 +1000
@@ -0,0 +1,7 @@
+# encoding: utf-8
+require_relative '<%= bootstrap_dir %>/unit'
+require '<%= src %>'
+
+class <%= testcase_name %>Test < MiniTest::Unit::TestCase
+  # TODO
+end
